rfc-drafts/stream.md

```diff
@@ -124,6 +124,52 @@ where
 }
 ```
 
+## Next method/struct
+
+We should also implement a next method, similar to [the implementation in the futures crate](https://docs.rs/futures-util/0.3.5/src/futures_util/stream/stream/next.rs.html#10-12).
```

Comment:
I think when we give the definition of the trait earlier, we should include something like

```rust
// Defined in std::stream module
pub trait Stream {
    // Core items:
    type Item;
    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>>;
    
    // Optional optimization hint, just like with iterators:
    #[inline]
    fn size_hint(&self) -> (usize, Option<usize>) {
        (0, None)
    }

    // Convenience methods (covered later on in the RFC):
    fn next(&mut self) -> Next<'_, Self>
    where
        Self: Unpin;
}
```

Reply:
I would also introduce this by saying "In general, we have purposefully kept the core trait definition minimal. There are a number of useful extension methods that are available, for example, in the futures-stream crate, but we have not included them because they involve closure arguments, and we have not yet finalized the design of async closures.

However, the core methods alone are extremely unergonomic. You can't even iterate over the items coming out of the stream. Therefore, we include a few minimal convenience methods that are not dependent on any unstable features. Most notably, `next`"

I guess I wrote this assuming we may add one or two more...

Reply:
Sounds good!